server/src/main/scala/controllers/Application.scala

```diff
@@ -584,6 +583,39 @@ class Application @Inject()(implicit val config: Configuration,
     }
   }
 
+  def exportApi(day: Int, month: Int, year: Int, terminalName: TerminalName) = auth {
+    Action.async { request =>
+      val dateOption = Try(SDate(year, month, day, 0, 0)).toOption
+      val terminalNameOption = airportConfig.terminalNames.find(name => name == terminalName)
+      val hasRole = ctrl.getRoles(config, request.headers, request.session).contains(ApiViewPortCsv)
```

Comment:
We can do this role restriction more elegantly for endpoints that don't need autowire, if you have a look at this example
https://github.com/UKHomeOffice/drt-v2/blob/master/server/src/main/scala/controllers/Application.scala#L759

We can add a similar function that takes roles as a curried parameter - something like this:
```
  def authRole[A](allowedRole: Role)(action: Action[A]) = Action.async(action.parser) { request =>
    if (loggedInUser.hasRole(allowedRole)) {
      Future(Unauthorized(s"{" +
        s"Permission denied, you need $allowedRole to access this page" +
        s"}"))
    } else {
      action(request)
    }
  }
```

Reply:
Then I think you can just wrap your action function with a call to the above 
```
def exportApi(day: Int, month: Int, year: Int, terminalName: TerminalName) = authByRole(ApiViewPortCsv) { auth {
...
}}
```